### PR TITLE
docs: update recommended strategy for models with input and output img and msk)

### PR DESCRIPTION
**Pull request by MathiasBaumgartinger**
Updates in the custom raster dataset tutorial and the actual file documentation. The previous recommended approach (overriding `__get_item__`) is outdated.

Refs: https://github.com/microsoft/torchgeo/discussions/2292

### CODE REVIEW

File: docs/tutorials/custom_raster_dataset.ipynb [hunk 1]
```diff
@@ -332,6 +332,10 @@
     "\n",
     "If your data only contains model inputs (such as images), use `is_image = True`. If your data only contains ground truth model outputs (such as segmentation masks), use `is_image = False` instead.\n",
     "\n",
+    "Consequently, the sample returned by the dataset/data loader will use the \"image\" key if *is_image* is True, otherwise it will use the \"mask\" key.\n",
+    "\n",
+    "For datasets with both model inputs and outputs, the recommended approach is to use 2 `RasterDataset` instances and combine them using a `IntersectionDataset`. See L7 Irish, L8 Biome, and I/O Bench for examples of this in torchgeo/datasets.\n",
```
Comment (adamjstewart) [inline docs/tutorials/custom_raster_dataset.ipynb:337]:
```suggestion
    "For datasets with both model inputs and outputs, the recommended approach is to use 2 `RasterDataset` instances and combine them using an `IntersectionDataset`. See L7 Irish, L8 Biome, and I/O Bench for examples of this in `torchgeo/datasets`.\n",
```

File: torchgeo/datasets/geo.py [hunk 2]
```diff
@@ -361,8 +361,8 @@ class RasterDataset(GeoDataset):
     #: The sample returned by the dataset/data loader will use the "image" key if
     #: *is_image* is True, otherwise it will use the "mask" key.
     #:
-    #: For datasets with both model inputs and outputs, a custom
-    #: :func:`~RasterDataset.__getitem__` method must be implemented.
+    #: For datasets with both model inputs and outputs, the recommended approach is
+    #: to use 2 `RasterDataset` instances and combine them using a `IntersectionDataset`.
```
Comment (adamjstewart) [inline torchgeo/datasets/geo.py:365]:
```suggestion
    #: to use 2 `RasterDataset` instances and combine them using an `IntersectionDataset`.
```

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
Thanks for the fix! Just waiting on the CLA now.

**Comment by adamjstewart** [comment]
@MathiasBaumgartinger can you accept the CLA?

**Comment by MathiasBaumgartinger** [comment]
@microsoft-github-policy-service agree

**Pull request merged**